src/test/java/it/gov/pagopa/payhub/activities/activity/classifications/IufClassificationActivityTest.java

```diff
@@ -0,0 +1,85 @@
+package it.gov.pagopa.payhub.activities.activity.classifications;
+
+import it.gov.pagopa.payhub.activities.dao.ClassificationDao;
+import it.gov.pagopa.payhub.activities.dao.PaymentsReportingDao;
+import it.gov.pagopa.payhub.activities.dto.classifications.Transfer2ClassifyDTO;
+import it.gov.pagopa.payhub.activities.dto.classifications.IufClassificationActivityResult;
+import it.gov.pagopa.payhub.activities.dto.paymentsreporting.PaymentsReportingDTO;
+import it.gov.pagopa.payhub.activities.utility.faker.PaymentsReportingFaker;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static it.gov.pagopa.payhub.activities.utility.faker.ClassifyResultFaker.buildTransfer2ClassifyDTO;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class IufClassificationActivityTest {
+    @Mock
+    private PaymentsReportingDao paymentsReportingDao;
+
+    @Mock
+    private ClassificationDao classificationDao;
+
+    private IufClassificationActivity iufClassificationActivity;
+
+    private static final Long ORGANIZATIONID = 1L;
+    private static final Long TREASURYID = 1L;
+    private static final String FLOWIDENTIFIERCODE = "FLOW";
+
+
+    @BeforeEach
+    void init() {
+        iufClassificationActivity = new IufClassificationActivityImpl(paymentsReportingDao, classificationDao);
+    }
+
+
```

Comment:
let's write an @AfterEach method to check that the all the invocations on the mock are verified (you will discover that the bug on the save operation when the paymentsReporting don't return record

see:
https://github.com/pagopa/p4pa-workflow-hub/blob/develop/src/test/java/it/gov/pagopa/pu/workflow/wf/ingestionflow/paymentsreporting/PaymentsReportingIngestionWFClientTest.java#L31

Reply:
added verify on mocks (also in the two tests)